CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+
+Kindly give your branch a more descriptive name like `docs-contributing-guide-2` instead of `patch-1`.
+
+You could follow this convention. Some ideas to get you started:
+
+- Feature Updates: `feat-<brief 2-3 words-Description>-<ISSUE_NO>`
+- Bug Fixes: `fix-<brief 2-3 words-Description>-<ISSUE_NO>`
+- Documentation: `docs-<brief 2-3 words-Description>-<ISSUE_NO>`
+- And so on...
+
+To create a new branch, use the following command:
+
+```bash
+git checkout -b your-branch-name
+```
+
+### Make the necessary changes.
+
+### Stage your changes and commit.
+
+```bash
+git add . # Stages all the changes
+git commit -m "<your_commit_message>"
+```
+
+Your commit message should be something which gives concise idea of the issue you are solving.
+
+We implement the Conventional Commits specification for commit messages. This specification is a lightweight convention on top of commit messages. It provides an easy set of rules for creating an explicit commit history; which makes it easier to write automated tools on top of. This convention dovetails with SemVer, by describing the features, fixes, and breaking changes made in commit messages.
+
+The commit message should be structured as follows:
+
+```bash
+<type>(optional scope): <description>
+```
+
+Push your local commits to the remote repository.
+
+```bash
+git push origin your-branch-name
+```
+
+**8.** Create a new [pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) from `your-branch-name`
+
+🎉 Congratulations! You've made your first pull request! Now, you should just wait until the maintainers review your pull request.
+
+## ✨ Understanding the project
+
+The project is divided into 3 main folders:
+
+- `packages` - This folder contains all the packages used within the million library
+- `test` - This folder contains all the tests written for the million library
+- `website` - This folder contains the code for the website [Millionjs](https://millionjs.org/)
+
+### `packages` folder
+
+The `packages` folder contains all the code related to the website. The website is built using [Next.js](https://nextjs.org/), a React framework. The website is hosted on [Vercel](https://vercel.com/).
+
+The source code is divided into two main folders:
+
+- `pages` - This folder contains all the pages of the website.
+- `components` - This folder contains all the components of the website.
+
+I suggest you to read the [Next.js documentation](https://nextjs.org/docs) to understand the project structure.
+
+To run the website locally, you need to install the dependencies and run the development server:
+
+```bash
+npm i -g pnpm # Install yarn
+pnpm install # Install dependencies
+pnpm {command} # Run the development server
+```
+
+
+`test` folder
+
+The `test` folder contains all the code related to the website. The website is built using [Next.js](https://nextjs.org/), a React framework. The website is hosted on [Vercel](https://vercel.com/).
```

Comment:
?

Reply:
This contributing guideline is a template that I am suggesting to be used for the project, The other things there can be edited later on to match the project's details, but, we should follow this template.

Reply:
Maybe you can help edit those parts that I copy-pasted several times to the appropriate details, so that I can commit them, then merge it to the repo.